Avoid cloning the toolbar in TabbedForm (#7350)
marmelab/react-admin: UPGRADE.md

Or in a component:
```diff

> export const PostEdit = (props) => (

Author: I change to a different example here to avoid confusion

-               invalid={invalid}
+               invalid={!isValid}
                saving={saving}
            />
        </Toolbar>
    );
};
```

### `submitOnEnter` Prop Has Been Removed

The following components no longer accept this prop:

- `<SimpleForm>`
- `<TabbedForm>`
- `<Toolbar>`
- `<SaveButton>`

> To determine wheter `<SimpleForm>` and `<TabbedForm>` will submit if `Enter` key is pressed, you have to relay on the `<SaveButton>`'s `type` prop. Setting it to "button" will prevent submittion on `Enter`.

Reviewer: ```suggestion
By default, `<SimpleForm>` and `<TabbedForm>` submit when the user presses `Enter`. To disable this behavior, you must now turn the `<SaveButton>` (which renders as a `<input type="submit" />` by default) into an `<input type="button">` element, by setting the `type` prop to "button".

If you didn't have a custom form toolbar, you'll have to create one to set the `<SaveButton type="button" />` and prevent submission on enter. 
```

};
```

Since the `<Toolbar>` component is not cloned anymore by react-admin layouts, properties like `mutationMode` and `submitOnEnter` are not passed down from parent components. So, if you rely on this behaviour, you should create your own custom toolbar instead

```diff
import { Toolbar, SimpleForm, Edit, TextInput } from 'react-admin';

> +const MyToolbar = props => <Toolbar {...props} mutationMode="pessimistic" submitOnEnter={false} />;

Reviewer 2: Actually, I think the Toolbar shouldn't clone and inject props either

Author: So, the solution should be to pass a custom SaveButton/DeleteButton inside a custom Toolbar?

};
```

> Since the `<Toolbar>` component is not cloned anymore by react-admin layouts, `submitOnEnter` property is not passed down to the `SaveButton` from parent components. So, if you rely on this behaviour, you should pass this prop yourself.

Reviewer: I think you should be more explicit and add a section titled 

> `<SimpleForm>` and `<TabbedForm>` don't support `submitOnEnter` anymore

and mention tat it also concerns Tabbedform

```diff
import { Toolbar, SimpleForm, Edit, TextInput, SaveButton, DeleteButton } from 'react-admin';

> +const MyToolbar = props => (
>     <Toolbar {...props}> 
>         <SaveButton submitOnEnter={false} />
>         <DeleteButton />
>     </Toolbar>
> );

Reviewer: ```suggestion
+const MyToolbar = props => (
+   <Toolbar {...props}> 
+       <SaveButton submitOnEnter={false} />
+       <DeleteButton />
+   </Toolbar>
+);
```

export const PostEdit = () => (
    <Edit>

> -       <SimpleForm>

Reviewer: ```suggestion
-       <SimpleForm submitOnEnter>
```

};
```

Since the `<Toolbar>` component is not cloned anymore by react-admin layouts, properties like `mutationMode` and `submitOnEnter` are not passed down from parent components. So, if you rely on this behaviour, you should create your own custom toolbar instead

```diff
import { Toolbar, SimpleForm, Edit, TextInput } from 'react-admin';

+const MyToolbar = props => <Toolbar {...props} mutationMode="pessimistic" submitOnEnter={false} />;

export const PostEdit = (props) => (

> -    <Edit {...props} mutationMode="pessimistic">

Author: `mutationMode` is used by `useEditController`, so is valid to pass it as a prop of an `Edit`/`EditBase` but it won't get to the toolbar's `DeleteButton`.
